### PR TITLE
AbstractFileSystem#listStatus returns a list of LocatedFileStatus instances

### PR DESCRIPTION
**Pull request by witgo**
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala#L315

### CODE REVIEW

**Comment by liuzx32** [comment]
@witgo Hello, Is the same question with the  issue https://github.com/Alluxio/alluxio/issues/9022 ?

**Comment by witgo** [comment]
@liuzx32  They're not the same bug.

**Comment by apc999** [comment]
@witgo  thanks for the patch. Can you elaborate here what's the issue we are fixing here? it is not clear to me by reading the link in this issue description

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 1]
```diff
@@ -52,6 +54,7 @@
 import org.apache.hadoop.fs.FSDataInputStream;
 import org.apache.hadoop.fs.FSDataOutputStream;
 import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.LocatedFileStatus;
```
Comment (apc999) [inline core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: which hadoop version introduces `org.apache.hadoop.fs.LocatedFileStatus`?
If it is a recent version, by having this line of src code, we might not be able to compile Alluxio client with earlier hadoop version. In this case, we may need to introduce multiple implementations of `AbstractFileSystem.java` for different Hadoop versions.

Comment (witgo) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: `hadoop 2.0 +` version introduces `org.apache.hadoop.fs.LocatedFileStatus`?

Comment (apc999) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: It looks like this is introduced in 2.x. So without your change @witgo , does this prevent you using any data source for Spark? If that is the case, please let us know and we will see the importance and address it accordingly.

**Pull request closed without merging**